### PR TITLE
Add new Message.reply method

### PR DESCRIPTION
**Pull request by bijij**
### Summary

Adds a new method `reply` to both Messsage and Context

Unsure if discord's implementation is final though so this implementation may need to change.

### Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by ChillFish8** [comment]
Pretty sure this was ruled out in [#5759](https://github.com/Rapptz/discord.py/issues/5759#issuecomment-681609438)
That Coroutine also seems rather pointless in the fact you:
- Cant change the reply message itself.
- Could just write that as a small helper function without adding extra things to the class.
- Really doesnt give much if any functionality over just using string formatting originally.

**Comment by TurnrDev** [comment]
Inspired by this PR and the reasons why this PR is declined, I'm working on a `quote` parameter. #5761 

My PR adds a parameter which just formats message content as a quote. It doesn't send a message, just returns a string to be used in another string. Because of this, it doesn't check if the quote is too long, I've left that up to the end-user/end developer.

**Comment by bijij** [comment]
@ChillFish8 unsure if this is the case as this is the actual discord reply feature Danny was referring to. It's currently locked behind a an experiment toggle however.

**Comment by AtieP** [comment]
Why do we really need this feature?

**Comment by numbermaniac** [comment]
This is a new feature being added to Discord. It already exists in Canary.

File: discord/message.py [hunk 1]
```diff
@@ -1073,3 +1073,35 @@ async def ack(self):
...
+        content: :class:`str`
+            The content of the message to send.
+        mention: :class:`bool`
+            Indicates if the reply should mention the message author, defaults to True.
+        **kwargs:
+            All keyword arguments for :meth:`~discord.abc.Messageable.send` can be used here.
+
+        Raises
+        --------
+        Exceptions raised by :meth:`~discord.abc.Messageable.send` are propegated by this function.
```
Comment (jack1142) [inline discord/message.py:1095]: A typo.
```suggestion
        Exceptions raised by :meth:`~discord.abc.Messageable.send` are propagated by this function.
```

**Comment by AtieP** [comment]
> This is a new feature being added to Discord. It already exists in Canary.

Ah, okay, I didn't know that.

File: discord/ext/commands/context.py [hunk 2]
```diff
@@ -322,3 +324,8 @@ async def send_help(self, *args):
                 return None
         except CommandError as e:
             await cmd.on_help_command_error(self, e)
+
+    async def reply(self, *args, **kwargs):
+        return self.message.reply(*args, **kwargs)
```
Comment (zephyrkul) [inline discord/ext/commands/context.py:329]:
```suggestion
        return await self.message.reply(*args, **kwargs)
```

Comment (SebbyLaw) [reply discord/ext/commands/context.py:329]: This change does nothing. The original code returned the un-awaited `Message.reply()` coroutine which is arguably better.

Comment (Phxntxm) [reply discord/ext/commands/context.py:329]: > This change does nothing. The original code returned the un-awaited `Message.reply()` coroutine which is arguably better.

But it in itself is a coroutine. Which means reply must be awaited, `await ctx.reply()` would then provide you with a coroutine... it needs to either be a coroutine and await the other coroutine, or not be a coroutine and return the coroutine.

**Comment by Herbstblatt** [comment]
This pr should probably be closed because discord decided to implement replies in other way and making new pr would be easier than updating that.

**Pull request closed without merging**